boa3/analyser/typeanalyser.py

```diff
@@ -26,10 +34,30 @@ def __init__(self, ast_tree: ast.AST):
 
         self.visit(self._tree)
 
-    def _log_error(self, error: CompilerError):
+    def _log_error(self, error: Error):
         self.errors.append(error)
         raise error
 
+    def get_type(self, value: Any) -> IType:
```

Comment:
Couldn't we have a method called get_type inside the Operation itself? So we don't need the IFs below

Reply:
forget about this. I read it wrong